prog/prio_test.go

```diff
@@ -59,3 +59,62 @@ func TestPrioChoice(t *testing.T) {
 		t.Fatal(diff)
 	}
 }
+
+// Test static priorities assigned based on the argument direction
+func TestStaticPriorities(t *testing.T) {
+	iters := 100000
+	target, err := GetTarget("linux", "amd64")
+	if err != nil {
+		t.Fatal(err)
+	}
+	// The first call is the one that creates a resource and the rest are calls that can use that resource.
+	tests := []string{
+		`r0 = open(&(0x7f0000393000-0x8)="2e2f66696c653000", 0x10000, 0x80)
+read(r0, &(0x7f0000617000)="00", 0x1)
+write(r0, &(0x7f0000617000+0x13)="2b03c3d04417ba11", 0x8)
+mmap(&(0x7f0000000000/0x1000)=nil, 0x1000, 0x3, 0x32, 0xffffffffffffffff, 0x0)`,
+		`r0 = socket(0x10, 0x2, 0x0)
+listen(r0, 0x1)
+setsockopt(r0, 0x4, 0x4, &(0x7f000056b000)="7a", 0x1)
+`,
+	}
+	for _, test := range tests {
+		p, err := target.Deserialize([]byte(test), Strict)
+		if err != nil {
+			t.Fatalf("failed to deserialize the program: %v", err)
+		}
+		enabled := make(map[*Syscall]bool)
+		for _, c := range p.Calls {
+			enabled[c.Meta] = true
+		}
+		ct := target.BuildChoiceTable(target.CalculatePriorities(nil), enabled)
```

Comment:
Does the test break if we don't use enabled?
It looks fishy if this breaks the test. And it doesn't then we could remove a bunch of code and build the table only once.

Reply:
Fixed